packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -950,3 +954,19 @@ function toSql(query: TemplateStringsArray | Sql, values: unknown[]): [Sql, Midd
 function isTemplateStringArray(value: unknown): value is TemplateStringsArray {
   return Array.isArray(value) && Array.isArray(value['raw'])
 }
+
+function getDataSourceOverrides(options: PrismaClientOptions | undefined, datasourceNames: string[]): Datasources {
+  if (!options) {
+    return {}
+  }
+
+  if (options.datasources) {
+    return options.datasources
+  }
+
+  if (options.datasourceUrl) {
+    const primaryDatasource = datasourceNames[0]
+    return { [primaryDatasource]: { url: options.datasourceUrl } }
+  }
+  return {}
+}
```

Comment:
Do you think it would make sense to move this to the `init` folder? I am trying to start organizing the constructor "stuff" in there.